keps/sig-node/4816-dra-prioritized-list/README.md

```diff
@@ -0,0 +1,1118 @@
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+This is an add-on on top of the `DynamicResourceAllocation` feature gate, which
+also must be enabled for this feature to work.
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DRAFirstAvailableOf
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-scheduler
+    - kube-controller-manager
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+Yes. No existing claims or running pods will be affected. This feature affects
+only the allocation of devices during scheduling.
+
+If a workload controller or Pod uses a `ResourceClaimTemplate` that includes
+this feature, it could happen that a new Pod may be created and need to be
+scheduled, even though the feature is disabled. In this case, the new Pod will
+fail to schedule, as the corresponding `ResourceClaim` will not be able to be
+created.
```

Comment:
Can we proscribe an approach to safely disable the feature on a cluster? Maybe finding all uses of `ResourceClaimTemplate` and taking an appropriate action for each?  (If a cluster admin needs to take action urgently to disable this feature, what steps can we recommend they take?)

Reply:
Delete all ResourceClaimTemplates and ResourceClaims, but that may be disruptive for workloads. depending on them. That's probably not the "safely" that you are looking for.

Reply:
I put a high level note describing this, let me know if you want more.

Reply:
This is what I was hoping for. Thanks!